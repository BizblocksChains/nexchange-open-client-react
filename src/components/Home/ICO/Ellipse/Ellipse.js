import React, { Component } from 'react';
import styles from './Ellipse.scss';

class Ellipse extends Component {
  state = {
    top: 0,
    left: 0,
  };

  componentDidMount = () => {
    setTimeout(this.updatePos, 0);
  };

  updatePos = () => {
    this.setState({
      top: 200 * (Math.random() > 0.5 ? -1 : 1),
      left: 200 * (Math.random() > 0.5 ? -1 : 1),
    });

    this.timeout = setTimeout(this.updatePos, Math.random() * (2500 - 1500) + 1500);
  };

  componentWillUnmount() {
    clearTimeout(this.timeout);
  }

  render() {
    return (
      <div className={`${styles.container} hidden-xs`} style={this.props.style}>
        <div
          className={styles.ellipse}
          style={{
            width: this.props.style.width,
            height: this.props.style.height,
            transform: `translate(${this.state.top}px, ${this.state.left}px)`,
          }}
        >
          <img src={require(`Img/ellipses/${this.props.coin}.svg`)} alt="LTC" />
        </div>
      </div>
    );
  }
}

export default Ellipse;
