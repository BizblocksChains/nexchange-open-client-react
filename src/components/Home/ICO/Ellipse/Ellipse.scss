.container {
  position: absolute;
  border-radius: 100%;
  z-index: 1;

  img {
    width: 43%;
    height: 43%;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
  }
}

.ellipse {
  display: inline-block;
  position: relative;
  background-image: url('Img/ellipse-full.svg');
  background-repeat: no-repeat;
  background-size: cover;

  -webkit-backface-visibility: hidden;
  -moz-backface-visibility: hidden;
  -ms-backface-visibility: hidden;
  backface-visibility: hidden;

  -webkit-perspective: 1000;
  -moz-perspective: 1000;
  -ms-perspective: 1000;
  perspective: 1000;

  transition: all 10000ms linear;
}
