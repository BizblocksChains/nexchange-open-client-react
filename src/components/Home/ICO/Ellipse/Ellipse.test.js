import React from 'react';
import { shallow } from 'enzyme';
import Ellipse from './Ellipse';

describe('Ellipse', () => {
  let wrapShallow;

  beforeEach(() => {
    wrapShallow = shallow(<Ellipse coin="btc" style={{ right: -20, left: 20, width: 48, height: 48 }} />);
  });

  it('renders correctly', () => {
    expect(wrapShallow).toMatchSnapshot();
  });

  it('clears interval on unmount', () => {
    const spy = jest.spyOn(window, 'clearTimeout');
    wrapShallow.unmount();
    expect(spy).toHaveBeenCalled();
  });

  it('updates position of top and left', () => {
    wrapShallow.instance().updatePos();

    expect(wrapShallow.state().top === 200 || wrapShallow.state().top === -200).toBeTruthy();
    expect(wrapShallow.state().left === 200 || wrapShallow.state().left === -200).toBeTruthy();
  });
});
