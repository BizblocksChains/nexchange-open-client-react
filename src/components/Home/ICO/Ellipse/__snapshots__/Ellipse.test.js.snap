// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ellipse renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Ellipse
    coin="btc"
    style={
        Object {
            "height": 48,
            "left": 20,
            "right": -20,
            "width": 48,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="ellipse"
        style={
                Object {
                        "height": 48,
                        "transform": "translate(0px, 0px)",
                        "width": 48,
                      }
        }
>
        <img
                alt="LTC"
                src="btc.svg"
        />
</div>,
      "className": "container hidden-xs",
      "style": Object {
        "height": 48,
        "left": 20,
        "right": -20,
        "width": 48,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          alt="LTC"
          src="btc.svg"
/>,
        "className": "ellipse",
        "style": Object {
          "height": 48,
          "transform": "translate(0px, 0px)",
          "width": 48,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "LTC",
          "src": "btc.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="ellipse"
          style={
                    Object {
                              "height": 48,
                              "transform": "translate(0px, 0px)",
                              "width": 48,
                            }
          }
>
          <img
                    alt="LTC"
                    src="btc.svg"
          />
</div>,
        "className": "container hidden-xs",
        "style": Object {
          "height": 48,
          "left": 20,
          "right": -20,
          "width": 48,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="LTC"
            src="btc.svg"
/>,
          "className": "ellipse",
          "style": Object {
            "height": 48,
            "transform": "translate(0px, 0px)",
            "width": 48,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "LTC",
            "src": "btc.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
